File: Java/DesignPatterns/Singleton_Design_pattern.java

// package Java.DesignPatterns;

class Logger{
    String filePath; 
    static Logger instance = null;
    private Logger(String filePath){
        System.out.println("object Created");
        this.filePath = filePath;
    }
    public static Logger getInstance(){

        System.out.println("Thread");
        if(instance == null)
            instance = new Logger("filePath");
        return instance;
    
    
    }
}


public class Singleton_Design_pattern {
    public static void main(String[] args) {
        Logger logger = Logger.getInstance();
        
        Runnable task = ()->{
            Logger.getInstance();
        };

  
        Thread thread1 = new Thread(task);
        Thread thread2 = new Thread(task);
        Thread thread3 = new Thread(task);
        
        thread1.start();
        thread2.start();
        thread3.start();
        

    

    }
}

File: JavaScript/DesignPatterns/Singleton_Design_pattern.js
class Logger {
    constructor(filePath) {
        console.log("object Created");
        this.filePath = filePath;
    }

    static getInstance() {
        console.log("Thread");
        if (Logger.instance == null) {
            Logger.instance = new Logger("filePath");
        }
        return Logger.instance;
    }
}
Logger.instance = null;

class Singleton_Design_pattern {
    static main(args) {
        const logger = Logger.getInstance();

        const task = () => {
            Logger.getInstance();
        };

        // Node.js has no native multi-threading for plain scripts; JS is
        // single-threaded, so the three "threads" are simulated as direct
        // sequential invocations to preserve the singleton behavior/output.
        const thread1 = task;
        const thread2 = task;
        const thread3 = task;

        thread1();
        thread2();
        thread3();
    }
}

Singleton_Design_pattern.main([]);
